server/src/main/java/io/crate/planner/Planner.java

```diff
@@ -235,6 +237,10 @@ public Plan plan(AnalyzedStatement analyzedStatement, PlannerContext plannerCont
         return analyzedStatement.accept(this, plannerContext);
     }
 
+    public BiFunction<LogicalPlan, PlannerContext, LogicalPlan> optimize() {
```

Comment:
I don't like exposing the optimize function like this, but I don't have a better idea. Any suggestions ?